src/components/Button/button.test.tsx

```diff
@@ -163,21 +163,50 @@ describe('<Button />', () => {
     });
   });
 
-  describe('when clicked', () => {
-    let clicked: boolean;
-    function clickMe() {
-      clicked = true;
-    }
+  describe('with event callbacks', () => {
+    let onClick: jest.Mock<{}>;
+    let onHoverIn: jest.Mock<{}>;
+    let onHoverOut: jest.Mock<{}>;
+
     beforeEach(() => {
-      clicked = false;  
+      onClick = jest.fn();  
+      onHoverIn = jest.fn();  
+      onHoverOut = jest.fn();  
       component = shallow(
-        <Button text={sampleText} onClick={clickMe} />,
+        <Button text={sampleText} onClick={onClick} onHoverIn={onHoverIn} onHoverOut={onHoverOut} />,
       );
     });
 
-    it('triggers its onClick callback', () => {
-      component.simulate('click');
-      expect(clicked).toBe(true);
+    describe('when clicked', () => {
+      beforeEach(() => {
+        component.simulate('click');
+      });
+
+      it('triggers its onClick callback', () => {
+        expect(onClick).toHaveBeenCalled();
+      });
+    });
+
+    describe('when hovered', () => {
+      beforeEach(() => {
+        component.simulate('mouseEnter');
+      });
+
+      it('triggers its onHoverIn callback', () => {
+        expect(onHoverIn).toHaveBeenCalled();
+        expect(onHoverOut).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('when hovered out', () => {
+      beforeEach(() => {
+        component.simulate('mouseLeave');
+      });
+
+      it('triggers its onHoverOut callback', () => {
+        expect(onHoverIn).not.toHaveBeenCalled();
```

Comment:
Testing onHoverIn/onHoverOut "not called" is not necessary. We're using onMouseLeave/onMouseEnter on the component, and you're testing what those native attributes do. Since one handler is passed, it isn't a valuable test for us.